draft-ietf-jsonpath-base.md

```diff
@@ -2239,6 +2259,34 @@ Other attacks can target the behavior of underlying technologies such as UTF-8 (
 
 --- back
 
+# Collected ABNF grammars
+
+This appendix collects the ABNF grammar from the ABNF passages used
+throughout the document.
+
+<!-- Update the collected grammar files using `make sourcecode`, which -->
+<!-- is currently manual as it creates a little circular dependency. -->
+<!-- The filenames of the ::includes are likely to change when -->
+<!-- kramdown-rfc-extract-sourcecode handles filenames better. -->
```

Comment:
With this branch, my workflow breaks. I normally issue `scripts/gen.sh`, but this now fails with:
```
 Created file draft-ietf-jsonpath-base.txt
 Created file draft-ietf-jsonpath-base.html
stdin(212:0): error: Rule jsonpath-query was already defined on line 0 of stdin
stdin(213:0): error: Rule segments was already defined on line 1 of stdin
stdin(214:0): error: Rule root-identifier was already defined on line 4 of stdin
...
stdin(389:0): error: Rule normal-index-selector was already defined on line 207 of stdin
parsing failed: 78 errors encountered
```
`make` however succeeds (after I installed npm), so maybe this doesn't matter.

Reply:
The problem with duplicating the ABNF is that ABNF doesn't accept duplicated rules (we learned from that in CDDL, which happily accepts [and checks!] duplicated rules).
We could change gen.sh to use sourcecode/abnf as well.

Reply:
> The problem with duplicating the ABNF is that ABNF doesn't accept duplicated rules (we learned from that in CDDL, which happily accepts [and checks!] duplicated rules).

Fair enough.

> We could change gen.sh to use sourcecode/abnf as well.

Yes please.

Reply:
> > The problem with duplicating the ABNF is that ABNF doesn't accept duplicated rules (we learned from that in CDDL, which happily accepts [and checks!] duplicated rules).
> 
> Fair enough.
> 
> > We could change gen.sh to use sourcecode/abnf as well.
> 
> Yes please.

Done.

Now we just have to remember to do a `make sourcecode` manually; you could add this to gen.sh but because of the circular dependency I'm refraining from that now.  (We also could put the `bap` checks in make sourcecode...  But then we require people to have bap installed.)